tests/post_training/pipelines/image_classification_timm.py

```diff
@@ -112,29 +120,35 @@ def transform_fn(data_item):
 
     def prepare_calibration_dataset(self):
         dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
-        loader = torch.utils.data.DataLoader(dataset, batch_size=1, num_workers=2, shuffle=False)
+        loader = torch.utils.data.DataLoader(dataset, batch_size=self.batch_size, num_workers=2, shuffle=False)
 
         self.calibration_dataset = nncf.Dataset(loader, self.get_transform_calibration_fn())
 
     def _validate(self):
         val_dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
-        val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=1, num_workers=2, shuffle=False)
-
-        dataset_size = len(val_loader)
-
-        predictions = [0] * dataset_size
-        references = [-1] * dataset_size
+        val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=self.batch_size, num_workers=2, shuffle=False)
+        dataset_size = len(val_dataset)
+        if dataset_size % self.batch_size != 0 and not self.dynamic_batch_shape:
+            raise ValueError(
+                (
+                    "Because the batch_size is not a divisor of the length of the dataset, "
+                    "the one of the data tensors has a shape incompatible with static model input. "
+                    "Use --dynamic_batch_shape option to export such model with dynamic shape."
+                )
+            )
 
         core = ov.Core()
+        ov_model = core.read_model(self.path_compressed_ir)
```

Comment:
Why it's moved upper? Are you sure that set_property applied for for compiled model?

Reply:
done